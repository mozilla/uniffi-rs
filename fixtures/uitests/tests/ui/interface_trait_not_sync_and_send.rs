// Unfortunately, path is relative to a temporary build directory :-/
uniffi_macros::generate_and_include_scaffolding!("../../../../fixtures/uitests/src/trait.udl");

fn main() { /* empty main required by `trybuild` */}

// This will fail to compile, because the trait is not explicit Send+Sync
pub trait Trait {
}

// This will fail to compile, because the trait is not explicit Send+Sync
#[uniffi::export]
pub trait ProcMacroTrait {
}
