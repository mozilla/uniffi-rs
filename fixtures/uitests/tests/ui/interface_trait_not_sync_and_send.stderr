error[E0277]: `dyn Trait` cannot be shared between threads safely
 --> $OUT_DIR[uniffi_uitests]/trait.uniffi.rs
  |
  | #[::uniffi::export_for_udl]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `dyn Trait` cannot be shared between threads safely
  |
  = help: the trait `Sync` is not implemented for `dyn Trait`
note: required by a bound in `_::{closure#0}::assert_impl_all`
 --> $OUT_DIR[uniffi_uitests]/trait.uniffi.rs
  |
  | #[::uniffi::export_for_udl]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
  = note: this error originates in the attribute macro `::uniffi::export_for_udl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `dyn Trait` cannot be sent between threads safely
 --> $OUT_DIR[uniffi_uitests]/trait.uniffi.rs
  |
  | #[::uniffi::export_for_udl]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `dyn Trait` cannot be sent between threads safely
  |
  = help: the trait `Send` is not implemented for `dyn Trait`
note: required by a bound in `_::{closure#0}::assert_impl_all`
 --> $OUT_DIR[uniffi_uitests]/trait.uniffi.rs
  |
  | #[::uniffi::export_for_udl]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
  = note: this error originates in the attribute macro `::uniffi::export_for_udl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `(dyn Trait + 'static)` cannot be shared between threads safely
 --> $OUT_DIR[uniffi_uitests]/trait.uniffi.rs
  |
  | #[::uniffi::export_for_udl]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn Trait + 'static)` cannot be shared between threads safely
  |
  = help: the trait `Sync` is not implemented for `(dyn Trait + 'static)`
note: required by a bound in `FfiConverterArc`
 --> $WORKSPACE/uniffi_core/src/ffi_converter_traits.rs
  |
  | pub unsafe trait FfiConverterArc<UT>: Send + Sync {
  |                                              ^^^^ required by this bound in `FfiConverterArc`
  = note: this error originates in the attribute macro `::uniffi::export_for_udl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `(dyn Trait + 'static)` cannot be sent between threads safely
 --> $OUT_DIR[uniffi_uitests]/trait.uniffi.rs
  |
  | #[::uniffi::export_for_udl]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn Trait + 'static)` cannot be sent between threads safely
  |
  = help: the trait `Send` is not implemented for `(dyn Trait + 'static)`
note: required by a bound in `FfiConverterArc`
 --> $WORKSPACE/uniffi_core/src/ffi_converter_traits.rs
  |
  | pub unsafe trait FfiConverterArc<UT>: Send + Sync {
  |                                       ^^^^ required by this bound in `FfiConverterArc`
  = note: this error originates in the attribute macro `::uniffi::export_for_udl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `dyn ProcMacroTrait` cannot be shared between threads safely
  --> tests/ui/interface_trait_not_sync_and_send.rs:11:1
   |
11 | #[uniffi::export]
   | ^^^^^^^^^^^^^^^^^ `dyn ProcMacroTrait` cannot be shared between threads safely
   |
   = help: the trait `Sync` is not implemented for `dyn ProcMacroTrait`
note: required by a bound in `_::{closure#0}::assert_impl_all`
  --> tests/ui/interface_trait_not_sync_and_send.rs:11:1
   |
11 | #[uniffi::export]
   | ^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
   = note: this error originates in the attribute macro `uniffi::export` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `dyn ProcMacroTrait` cannot be sent between threads safely
  --> tests/ui/interface_trait_not_sync_and_send.rs:11:1
   |
11 | #[uniffi::export]
   | ^^^^^^^^^^^^^^^^^ `dyn ProcMacroTrait` cannot be sent between threads safely
   |
   = help: the trait `Send` is not implemented for `dyn ProcMacroTrait`
note: required by a bound in `_::{closure#0}::assert_impl_all`
  --> tests/ui/interface_trait_not_sync_and_send.rs:11:1
   |
11 | #[uniffi::export]
   | ^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
   = note: this error originates in the attribute macro `uniffi::export` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `(dyn ProcMacroTrait + 'static)` cannot be shared between threads safely
  --> tests/ui/interface_trait_not_sync_and_send.rs:11:1
   |
11 | #[uniffi::export]
   | ^^^^^^^^^^^^^^^^^ `(dyn ProcMacroTrait + 'static)` cannot be shared between threads safely
   |
   = help: the trait `Sync` is not implemented for `(dyn ProcMacroTrait + 'static)`
note: required by a bound in `FfiConverterArc`
  --> $WORKSPACE/uniffi_core/src/ffi_converter_traits.rs
   |
   | pub unsafe trait FfiConverterArc<UT>: Send + Sync {
   |                                              ^^^^ required by this bound in `FfiConverterArc`
   = note: this error originates in the attribute macro `uniffi::export` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `(dyn ProcMacroTrait + 'static)` cannot be sent between threads safely
  --> tests/ui/interface_trait_not_sync_and_send.rs:11:1
   |
11 | #[uniffi::export]
   | ^^^^^^^^^^^^^^^^^ `(dyn ProcMacroTrait + 'static)` cannot be sent between threads safely
   |
   = help: the trait `Send` is not implemented for `(dyn ProcMacroTrait + 'static)`
note: required by a bound in `FfiConverterArc`
  --> $WORKSPACE/uniffi_core/src/ffi_converter_traits.rs
   |
   | pub unsafe trait FfiConverterArc<UT>: Send + Sync {
   |                                       ^^^^ required by this bound in `FfiConverterArc`
   = note: this error originates in the attribute macro `uniffi::export` (in Nightly builds, run with -Z macro-backtrace for more info)
